server/api.go

```diff
@@ -92,6 +97,28 @@ func (p *Plugin) userConnected(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func (p *Plugin) userWantsToGiveFeedback(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+
+	post, err := p.CreateBotDMPost(userID, &model.Post{
+		Message: feedbackRequestBody,
+		Type:    "custom_nps_feedback",
+	})
+	if err != nil {
+		p.API.LogError("Failed to send user the give feedback message", "user_id", userID, "err", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json")
+	if err := json.NewEncoder(w).Encode(post); err != nil {
+		p.API.LogWarn("unable to encode feedback message post", "err", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	w.WriteHeader(http.StatusOK)
```

Comment:
If I'm reading correctly, this is missing writing the encoded json to the response.

Reply:
A few lines above, we encode directly into the ResponseWriter 

`json.NewEncoder(w).Encode(post)`

Reply:
Ahh, I wasn't familiar with notation! Makes sense now, and it looks common across `mattermost-server`